Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DPEComponentAdapter.cpp

```diff
@@ -87,4 +88,54 @@ namespace AZ::DocumentPropertyEditor
         NotifyResetDocument();
     }
 
+    Dom::Value ComponentAdapter::HandleMessage(const AdapterMessage& message)
+    {
+        auto handlePropertyEditorChanged = [&]([[maybe_unused]] const Dom::Value& valueFromEditor, Nodes::ValueChangeType changeType)
+        {
+            switch (changeType)
+            {
+            case Nodes::ValueChangeType::InProgressEdit:
+                if (m_componentInstance)
+                {
+                    const AZ::EntityId& entityId = m_componentInstance->GetEntityId();
+                    if (entityId.IsValid())
```

Comment:
Is this even possible?

Reply:
Do you mean is it possible for the EntityId to be invalid? If so, I'm not entirely sure but it seemed like a good thing to check.

Reply:
Actually, I see in EntityId Component::GetEntityId() const that it will return the InvalidEntityId if its m_entity is nulltpr.

Reply:
Yes, I'm sure it can be invalid logically. I'm just wondering whether it is possible for a component adapter's component to have an invalid entity id once it is created. It just seems like an impossible scenario. But the logic can stay as an additional guardrail